gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpRequestRunnable.java

```diff
@@ -109,30 +110,32 @@ public void run() {
       }
 
       // Populate HTTP path and query parameters.
-      Map<String, String> pathParams =
-          requestBuilder.getPathParams(request, methodDescriptor.getResourceNameField());
-      PathTemplate pathPattern = PathTemplate.create(methodDescriptor.endpointPathTemplate());
-      String relativePath = pathPattern.instantiate(pathParams);
-      GenericUrl url = new GenericUrl(endpoint + relativePath);
-      Map<String, List<String>> queryParams =
-          requestBuilder.getQueryParams(request, methodDescriptor.getQueryParams());
-      for (String queryParam : methodDescriptor.getQueryParams()) {
-        if (queryParams.containsKey(queryParam) && queryParams.get(queryParam) != null) {
-          url.set(queryParam, queryParams.get(queryParam));
+      GenericUrl url = new GenericUrl(endpoint + requestFormatter.getEndpointRelativePath(request));
+      Map<String, List<String>> queryParams = requestFormatter.getQueryParams(request);
+      for (String queryParam : queryParams.keySet()) {
+        if (queryParams.get(queryParam) != null) {
+          for (String val : queryParams.get(queryParam)) {
```

Comment:
If you need both the key & value, then iterate over Map.Entry instead of just the keys in the outer loop.

Reply:
Done.